### PR TITLE
Disable list commits toggle when there are only root branches

### PR DESCRIPTION
**Pull request by PawelLipski**
image

### CODE REVIEW

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/ToggleListCommitsAction.java [hunk 1]
```diff
@@ -14,12 +15,26 @@
...
+  @Override
+  @UIEffect
+  public void update(AnActionEvent e) {
+    super.update(e);
+
+    var branchLayout = getBranchLayout(e);
+    if (branchLayout.isDefined()) {
+      boolean anyChildBranchExists = branchLayout.get().getRootEntries()
+          .exists(rootBranch -> rootBranch.getSubentries().nonEmpty());
+      e.getPresentation().setEnabled(anyChildBranchExists);
```
Comment (mkondratek) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/ToggleListCommitsAction.java:34]: Hmmm :thinking: what about setting some description that will explain to the user why the action is disabled?

Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/ToggleListCommitsAction.java:34]: Done

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/ToggleListingCommitsAction.java [hunk 2]
```diff
@@ -14,12 +15,33 @@
...
+    if (branchLayout.isDefined()) {
+      boolean anyChildBranchExists = branchLayout.get().getRootEntries()
+          .exists(rootBranch -> rootBranch.getSubentries().nonEmpty());
+      var presentation = e.getPresentation();
+      if (anyChildBranchExists) {
+        presentation.setEnabled(true);
+        presentation.setDescription("Toggle listing commits");
+      } else {
+        presentation.setEnabled(false);
+        presentation.setDescription("Toggle listing commits disabled: no child branches present");
```
Comment (mkondratek) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/ToggleListingCommitsAction.java:40]: :heart:

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -1,15 +1,14 @@
 <idea-plugin>
     <id>com.virtuslab.git-machete</id>
     <name>Git Machete</name>
-    <vendor email="gitmachete@virtuslab.com" url="http://virtuslab.com">VirtusLab</vendor>
+    <vendor email="gitmachete@virtuslab.com" url="https://virtuslab.com">VirtusLab</vendor>
```
Comment (mkondratek) [inline src/main/resources/META-INF/plugin.xml:4]: Good point!

File: src/main/resources/META-INF/plugin.xml [hunk 4]
```diff
@@ -24,45 +23,59 @@
...
+                <override-text place="GitMacheteContextMenu" text="_Check Out Branch"/>
+            </action>
 
             <action id="GitMachete.RebaseSelectedBranchOntoParentAction"
                     class="com.virtuslab.gitmachete.frontend.actions.RebaseSelectedBranchOntoParentAction"
-                    text="_Rebase Branch Onto Parent"
-                    description="Rebase selected branch onto parent"/>
+                    text="Git Machete: Rebase Selected Branch Onto Parent"
+                    description="Rebase this branch onto parent">
+                <override-text place="GitMacheteContextMenu" text="_Rebase Branch Onto Parent"/>
```
Comment (mkondratek) [inline src/main/resources/META-INF/plugin.xml:35]: rocket-ish!  :rocket:

**Pull request merged**